chain/chunks/src/lib.rs

```diff
@@ -1771,50 +1776,92 @@ impl ShardsManager {
             owned_parts,
         );
 
+        let protocol_version = self.epoch_manager.get_epoch_protocol_version(epoch_id)?;
         let block_producers =
             self.epoch_manager.get_epoch_block_producers_ordered(&epoch_id, lastest_block_hash)?;
         let current_chunk_height = partial_encoded_chunk.header.height_created();
-        let mut next_chunk_producers = self
-            .epoch_manager
-            .shard_ids(&epoch_id)?
-            .into_iter()
-            .map(|shard_id| {
-                self.epoch_manager.get_chunk_producer(&epoch_id, current_chunk_height + 1, shard_id)
-            })
-            .collect::<Result<HashSet<_>, _>>()?;
-        next_chunk_producers.remove(me);
-        for (bp, _) in block_producers {
-            let bp_account_id = bp.take_account_id();
-            // no need to send anything to myself
-            if me == &bp_account_id {
-                continue;
+
+        if checked_feature!("stable", SingleShardTracking, protocol_version) {
+            let shard_id = partial_encoded_chunk.header.shard_id();
+            let mut accounts_forwarded_to = HashSet::new();
+            accounts_forwarded_to.insert(me.clone());
+            let next_chunk_producer = self.epoch_manager.get_chunk_producer(
+                &epoch_id,
+                current_chunk_height + 1,
+                shard_id,
+            )?;
+            for (bp, _) in block_producers {
+                let bp_account_id = bp.take_account_id();
+
+                if cares_about_shard_this_or_next_epoch(
+                    Some(&bp_account_id),
+                    lastest_block_hash,
```

Comment:
```suggestion
                    latest_block_hash,
```
As you change it anyway :)